### PR TITLE
Internal: Editor V2 - Fix `typescript-eslint` rules [ED-9485]

### PR DESCRIPTION
**Pull request by StyleShit**

### CODE REVIEW

File: packages/packages/v1-adapters/src/dispatchers/utils.ts [hunk 1]
```diff
@@ -1,10 +1,14 @@
 import { jQueryDeferred } from './types';
 
-export function isJQueryDeferred<T>( value: any ): value is jQueryDeferred<T> {
+export function isJQueryDeferred<T>( value: unknown ): value is jQueryDeferred<T> {
 	// TODO: Copied from:
 	//  https://github.com/elementor/elementor/blob/6a74fc9/modules/web-cli/assets/js/core/commands.js#L410
 
-	return value && 'object' === typeof value && value.promise && value.then && value.fail;
+	return !! value &&
```
Comment (rotemee) [inline packages/packages/v1-adapters/src/dispatchers/utils.ts:7]: Consider adding parenthesis.

**Review by rotemee — CHANGES_REQUESTED** [review]

**Review by rotemee — DISMISSED** [review]

File: packages/packages/v1-adapters/src/dispatchers/__tests__/index.test.ts [hunk 2]
```diff
@@ -1,29 +1,41 @@
...
 
+type ExtendedWindow = Window & {
+	$e: {
+		run: jest.Mock;
+		route: jest.Mock;
+	}
+}
+
 describe( '@elementor/v1-adapters/dispatchers', () => {
+	let eRun: jest.Mock, eRoute: jest.Mock;
```
Comment (Nevoss) [inline packages/packages/v1-adapters/src/dispatchers/__tests__/index.test.ts:11]:
```suggestion
	let eRun: jest.Mock,
		eRoute: jest.Mock;
```

**Review by Nevoss — CHANGES_REQUESTED** [review]

File: packages/packages/v1-adapters/src/dispatchers/__tests__/index.test.ts (hunk 2 above)
Comment (StyleShit) [reply packages/packages/v1-adapters/src/dispatchers/__tests__/index.test.ts:11]: Done

File: packages/packages/v1-adapters/src/listeners/__tests__/index.test.ts [hunk 3]
```diff
@@ -141,7 +141,11 @@ describe( '@elementor/v1-adapters/listeners', () => {
 		// Arrange.
 		const callback = jest.fn();
 
-		( window as any ).__elementorEditorV1LoadingPromise = new Promise( ( resolve ) => {
+		type ExtendedWindow = Window & {
```
[the remark this answers is not in the document]
Comment (StyleShit) [reply packages/packages/v1-adapters/src/listeners/__tests__/index.test.ts:328]: I don't want other packages to know about global variables they shouldn't know

[the remark this answers is not in the document]
Comment (Nevoss) [reply packages/packages/v1-adapters/src/listeners/__tests__/index.test.ts:328]: What about internal use?

**Review by Nevoss — APPROVED** [review]

**Pull request merged**